phoenix-core/src/main/java/org/apache/phoenix/execute/PhoenixTxIndexMutationGenerator.java

```diff
@@ -364,7 +364,8 @@ private boolean generatePuts(
                 throw new IOException(e);
             }
         }
-        Iterable<IndexUpdate> puts = codec.getIndexUpserts(state, indexMetaData, regionStartKey, regionEndKey);
+        Iterable<IndexUpdate> puts =
+                codec.getIndexUpserts(state, indexMetaData, regionStartKey, regionEndKey, true);
```

Comment:
Shouldn't verified be false here? Are we generating index upsert mutations in verified=true even though the data table write fails? A code comment would be nice.

Reply:
@gokceni - I believe the purpose is to allow the old "direct" index tool mapper to be able to build new-style indexes, since snapshot-based and transactional index builds still need to use the direct mapper. Since it's reading existing base table data, we can assume the new index write is verified.

Reply:
This is in the normal write path via PhoenixTransactionalIndexer, not the in the async index generation.

Three-phase commit is not used in the normal TX write path, because the transactionality is taking care of the
consistency, and accordingly it never updates the status to VERIFIED, and leaves the status as "x".

This method is also used by the old indexing implementation, and I didn't want old-style tables to set the status to verified, hence the new parameter.

The same basic problem pattern that BulkLoad and Async Indextool has, but this one is in the normal write path.

I should have called this change out explicitly, as admittedly this has the bigggest impact in the patch, but forgot, sorry.

This really should have been four patches (snapshot, Async VERIFIED, TX index status, and the directApi purge), but I got carried away as I kept finding the issues by the expanded test case.